File: ActivityFinder.Server/Models/Activity/Activity.cs
namespace ActivityFinder.Server.Models
{
    public class Activity
    {
        public int ActivityId { get; set; }
        public required string Title { get; set; }
        public required string Description { get; set; }
        //public required string Category { get; set; }
        public required DateTime Date { get; set; }
        public string? OtherInfo { get; set; }

        public DbProperties DbProperties { get; set; } = new DbProperties();

        public int AddressId { get; set; }
        public required Address Address { get; set; }

        public string CreatorId { get; set; }
        public required ApplicationUser Creator { get; set; }

        public List<ApplicationUser> JoinedUsers { get; set; } = [];

    }
}

File: activityfinder.client/src/app/interfaces/activity.ts
import { Address } from "./address";

export interface Activity {
  id: number;
  description: string;
  title: string;
  category: string;
  date: Date;
  otherInfo: string;

  address: Address;
  displayName: string;
}

export interface ActivityListItem {
  id: number;
  date: Date;
  title: string;
  address: string;
}
